ui/app/mixins/cluster-route.js

```diff
@@ -70,6 +71,12 @@ export default Mixin.create({
       return UNSEAL;
     }
     if (get(cluster, 'dr.isSecondary')) {
+      if (transition && transition.targetName === DR_REPLICATION_SECONDARY_DETAILS) {
```

Comment:
You could do `transition && transition.targetName.startsWith(DR_REPLICATION_SECONDARY)` here and then return `transition.targetName` below if you wanted, then you'd only need one `if`.

Reply:
Nice!  Thank you.  I know you had mentioned `startsWith` before when we spoke, but I didn't see quite how to use it.  The return transition.tragetName is perfect.  Thanks!!

Reply:
OK, in testing, I found a problem with this approach: it does not auto-refresh when turning a cluster into a DR secondary.  

Gif of not auto-refreshing under the suggested solution that Matthew gave: 
<img width=400 src="https://user-images.githubusercontent.com/6618863/78066141-a4a45600-7351-11ea-82da-f220fb0213e5.gif">

The reason, at least that I can see, that it doesn't auto-refresh in this condition is because it gets stuck in a weird state after the transition to `vault.cluster.unseal`.  The only way I was able to get it to work (aside from my first two if statements) was as follows:

```
if (transition && !transition.targetName) { 
    return DR_REPLICATION_SECONDARY;
 }
 if (transition && transition.targetName.startsWith(DR_REPLICATION_SECONDARY)) {  
     return transition.targetName;
  }
  return DR_REPLICATION_SECONDARY;
```
This completely 🤯 because the first if statement shouldn't be necessary as that situation _should_ be caught in the last `return DR_REPLICATION_SECONDARY`.  I think some funky state over-ride is happening during the transition, but I can't say for certain.  

Suffice it to say, I'm going to argue for my original two if statements solution because it contains the change to something very specific (the details page) therefore having less ability to affect other refresh states as seen here.  It also does not have this specific bug, or any other that I was able to test for. Thoughts?

Reply:
OK, in testing, I found a problem with this approach: it does not auto-refresh when turning a cluster into a DR secondary.  

Gif of not auto-refreshing under the suggested solution that Matthew gave: 
<img width=400 src="https://user-images.githubusercontent.com/6618863/78066141-a4a45600-7351-11ea-82da-f220fb0213e5.gif">

The reason, at least that I can see, that it doesn't auto-refresh in this condition is because it gets stuck in a weird state after the transition to `vault.cluster.unseal`.  The only way I was able to get it to work (aside from my first two if statements) was as follows:

```
if (transition && !transition.targetName) { 
    return DR_REPLICATION_SECONDARY;
 }
 if (transition && transition.targetName.startsWith(DR_REPLICATION_SECONDARY)) {  
     return transition.targetName;
  }
  return DR_REPLICATION_SECONDARY;
```
This completely 🤯 because the first if statement shouldn't be necessary as that situation _should_ be caught in the last `return DR_REPLICATION_SECONDARY`.  I think some funky state over-ride is happening during the transition, but I can't say for certain.  

Suffice it to say, I'm going to argue for my original two if statements solution because it contains the change to something very specific (the details page) therefore having less ability to affect other refresh states as seen here.  It also does not have this specific bug, or any other that I was able to test for.  @meirish Thoughts?

Reply:
Oof yeah that's really odd 🤔 - totally agree on moving back to the original though!